hoomd/mpcd/ConfinedStreamingMethod.h

```diff
@@ -128,11 +129,16 @@ void ConfinedStreamingMethod<Geometry>::stream(unsigned int timestep)
 
         // propagate the particle to its new position ballistically
         Scalar dt_remain = m_mpcd_dt;
+        mask_collision.data[cur_p] = 0; //if collide is true this value will get changed to 1, otherwise will stay 0
         bool collide = true;
         do
             {
             pos += dt_remain * vel;
             collide = m_geom->detectCollision(pos, vel, dt_remain);
+            if (collide)
+                {
+                mask_collision.data[cur_p] = 1; //if collide is true, it will be 1
+                }
```

Comment:
```suggestion
            bounced |= collide;
```

Then, you should write this out once after the loop is done:
```
h_bounced.data[cur_p] = bounced;
```